util/processinfo.go

```diff
@@ -161,9 +161,9 @@ func serverStatus2Str(state uint16) string {
 // SessionManager is an interface for session manage. Show processlist and
 // kill statement rely on this interface.
 type SessionManager interface {
-	ShowProcessList() map[uint64]*ProcessInfo
+	ShowProcessList(func(*ProcessInfo))
 	ShowTxnList() []*txninfo.TxnInfo
-	GetProcessInfo(id uint64) (*ProcessInfo, bool)
+	GetProcessInfo(id uint64, f func(*ProcessInfo))
```

Comment:
1. Maybe `IterProcessInfo` or `VisitProcessInfo..`. A better name.
2. Replace two APIs with one. `ShowProcessList(func(*ProcessInfo) bool)`, for `GetProcessInfo`, user could use `id == xxx` to judge, and return false to stop the iteration.

Reply:
2. provide the `id` parameter is necessary, because if we do not provide it and let the caller use `id == xxx` to judge, the algorithm is O(n)